lib/action_controller/responder.rb

```diff
@@ -182,7 +182,9 @@ def to_js
     # responds to :to_format and display it.
     #
     def to_format
-      if get? || !has_errors? || response_overridden?
+      if ActionPack::VERSION::MAJOR >= 5 && controller.is_a?(ActionController::API)
```

Comment:
can we replace the `ActionPack::VERSION::MAJOR` with `defined?(ActionController::API)`? This way we aren't strictly bound to version numbers for the the module's definition.